module/skill/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,382 @@
+        Eigen::Vector3f thetaST = Eigen::Vector3f::Zero();
+        // Position of flying foot {f} relative to support foot {s}
+        Eigen::Vector3f rFSs = Eigen::Vector3f::Zero();
+        // Euler angles [Roll, Pitch, Yaw] of flying foot {f} relative to support foot {s}
+        Eigen::Vector3f thetaSF = Eigen::Vector3f::Zero();
+        // Read the cartesian positions and orientations for trunk and fly foot
+        std::tie(rTSs, thetaST, rFSs, thetaSF, is_left_support) = walk_engine.compute_cartesian_position();
+
+        // Change goals from support foot based coordinate system to trunk based coordinate system
+        // Trunk {t} from support foot {s}
+        Eigen::Isometry3f Hst;
+        Hst.linear()      = EulerIntrinsicToMatrix(thetaST);
+        Hst.translation() = rTSs;
+
+        // Flying foot {f} from support foot {s}
+        Eigen::Isometry3f Hsf;
+        Hsf.linear()      = EulerIntrinsicToMatrix(thetaSF);
+        Hsf.translation() = rFSs;
+
+        // Support foot {s} from trunk {t}
+        const Eigen::Isometry3f Hts = Hst.inverse();
+
+        // Flying foot {f} from trunk {t}
+        const Eigen::Isometry3f Htf = Hts * Hsf;
+
+        // Get desired transform for left foot {l}
+        const Eigen::Isometry3f Htl = walk_engine.get_footstep().is_left_support() ? Hts : Htf;
+
+        // Get desired transform for right foot {r}
+        const Eigen::Isometry3f Htr = walk_engine.get_footstep().is_left_support() ? Htf : Hts;
+
+        // ****Create limb tasks****
+        const NUClear::clock::time_point time = NUClear::clock::now() + Per<std::chrono::seconds>(UPDATE_FREQUENCY);
+
+        //  Legs
+        auto left_leg   = std::make_unique<LeftLegIK>();
+        auto right_leg  = std::make_unique<RightLegIK>();
+        left_leg->time  = time;
+        right_leg->time = time;
+        left_leg->Htl   = Htl.cast<double>().matrix();
+        right_leg->Htr  = Htr.cast<double>().matrix();
+        // Arms
+        auto left_arm  = std::make_unique<LeftArm>();
+        auto right_arm = std::make_unique<RightArm>();
+
+        // Loop to set the servo states
+        for (int id = 0; id < ServoID::NUMBER_OF_SERVOS - 2; ++id) {
+            // Set the legs
+            if ((id >= ServoID::R_HIP_YAW) && (id % 2 == 0)) {  // right legs
+                right_leg->servos[id] = current_cfg.jointGains[ServoID(id)]
+            }
+            else if ((id >= ServoID::R_HIP_YAW) && (id % 2 == 1)) {  // left legs
+
+                left_leg->servos[id] = current_cfg.jointGains[ServoID(id)];
+            }
+            else if ((id < ServoID::R_HIP_YAW) && (id % 2 == 0)) {  // right arms
+                right_arm->servos[id] = ServoCommand(time,
+                                                     current_cfg.arm_positions[ServoID(id)].second,
+                                                     ServoState(current_cfg.jointGains[ServoID(id)], 100));
```

Comment:
```suggestion
                                                     current_cfg.jointGains[ServoID(id)]);
```
should also rename this variable to something like `joint_servostate`